serving-api/src/main/scala/com/stratio/sparkta/serving/api/actor/ClusterLauncherActor.scala

```diff
@@ -126,25 +100,49 @@ class ClusterLauncherActor(policy: AggregationPoliciesModel, policyStatusActor:
     } else false
 
   private def launch(main: String, hdfsDriverFile: String, master: String, args: Map[String, String],
-                     driverParams: Seq[String]): Unit = {
+                     driverParams: Seq[String], clusterFiles: Seq[String]): Unit = {
     val sparkLauncher = new SparkLauncher()
       .setSparkHome(sparkHome)
       .setAppResource(hdfsDriverFile)
       .setMainClass(main)
       .setMaster(master)
+    clusterFiles.foreach(file => sparkLauncher.addJar(file))
     args.map({ case (k: String, v: String) => sparkLauncher.addSparkArg(k, v) })
     if (isStandaloneSupervise) sparkLauncher.addSparkArg(StandaloneSupervise)
     //Spark params (everything starting with spark.)
-    sparkConf.map({ case (key: String, value: String) => sparkLauncher.setConf(key, value) })
+    sparkConf.map({ case (key: String, value: String) =>
+      sparkLauncher.setConf(key, if(key == "spark.app.name")  s"$value-${policy.name}" else value)
+    })
     // Driver (Sparkta) params
     driverParams.map(sparkLauncher.addAppArgs(_))
-    val spark = sparkLauncher.launch()
-    val loggerThread = new ClusterLogger(spark)
-    loggerThread.run()
-    spark.waitFor()
-    val exit = spark.exitValue()
-    log.debug(s"Spark exit status: $exit")
-    if (exit != 0) setErrorStatus
+
+    log.info(s"$sparkHome/bin/spark-submit --master $master " ++ args.foldLeft("") {
+      (s: String, pair: (String, String)) =>  s"$s ${pair._1} ${pair._2}"
+    })
+
+    Future[(Int, Process)] {
```

Comment:
Is it necessary that this block of code is asynchronous?. I prefer not to use futures here (is is only my personal opinion).